src/lazy/expanded/sequence.rs

```diff
@@ -212,8 +211,6 @@ pub enum ExpandedSExpSource<'top, D: Decoder> {
     /// Because the TDL treats s-expressions as literals, this variant only applies when the
     /// s-expression appeared within a `literal` operation.
     Template(Environment<'top, D>, TemplateElement<'top>),
-    /// The SExp was produced by a call to `make_sexp`.
-    Constructed(Environment<'top, D>, MacroExprArgsIterator<'top, D>),
```

Comment:
Defining `make_sexp` in terms of `flatten` allowed me to delete lots of special-case code for `make_sexp`. 🎉